src/main/java/com/spotify/futures/ExecutionMetadata.java

```diff
@@ -0,0 +1,67 @@
+package com.spotify.futures;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+/**
+ * Provides execution metadata in relation to a group of completable futures.
+ *
+ * @author Jose Alavez
+ * @since 0.3.3
+ */
+public class ExecutionMetadata {
+
+  private final CompletableFuture<?>[] all;
+
+  ExecutionMetadata(CompletableFuture<?>... all) {
+    this.all = all;
+  }
+
+  private int accumulate(Function<CompletableFuture<?>, Boolean> fn) {
```

Comment:
rename to countMatches or something similar

Reply:
- [ ] Rename